janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/PartitionedVertexProgramExecutor.java

```diff
@@ -80,6 +80,15 @@ public void run(int numThreads, ScanMetrics metrics) {
         }
     }
 
+    private StandardJanusGraphTx startTransaction(StandardJanusGraph graph) {
```

Comment:
nitpick: Would it a good idea to move startTransaction into the AbstractScanJob.

Reply:
But this class PartitionedVertexProgramExecutor.java is not and cannot be a subclass of AbstractScanJob.

Reply:
@li-boxuan What if we just move it to `StandardJanusGraphTx` itself? I'm just thinking about removing the duplication code.
Something like (not sure about method name):
```
    public static StandardJanusGraphTx startNoCacheNoInternalCheckPreloadedDataTransaction(StandardJanusGraph graph) {
        StandardTransactionBuilder txb = graph.buildTransaction().readOnly();
        txb.setPreloadedData(true);
        txb.checkInternalVertexExistence(false);
        txb.dirtyVertexSize(0);
        txb.vertexCacheSize(0);
        return (StandardJanusGraphTx)txb.start();
    }
```
If so, we can use this static method in both PartitionedVertexProgramExecutor and VertexJobConverter. 
I'm also good moving this method to a separate utility class to not mess `StandardJanusGraphTx` with such methods.

Reply:
I moved it to StandardTransactionBuilder class.